### ISSUE TITLE
--dockerfile flag not working

### ISSUE THREAD

**Issue by elliotdickison** [superfly/flyctl#400]
**Describe the bug**

The `flyctl deploy` command's `--dockerfile` flag doesn't appear to be working for me. I have a monorepo setup (directory structure below) but when I run `flyctl deploy . --dockerfile ./packages/client/Dockerfile --config ./packages/client/fly.toml` from the `app` directory I get an error ending with `Dockerfile: no such file or directory`

```
app/
├─ packages/
│  ├─ client/ 
│  │  ├─ fly.toml
│  │  ├─ Dockerfile
│  ├─ server/ 
│  │  ├─ fly.toml
│  │  ├─ Dockerfile
├─ package.json
├─ yarn.lock
```

**Add `fly.toml`**

```toml
# fly.toml file generated for icy-sky-8224 on 2021-04-19T16:44:17-07:00

app = "icy-sky-8224"

kill_signal = "SIGINT"
kill_timeout = 5

[[services]]
  http_checks = []
  internal_port = 3000
  protocol = "tcp"
  script_checks = []

  [services.concurrency]
    hard_limit = 25
    soft_limit = 20
    type = "connections"

  [[services.ports]]
    handlers = ["http"]
    port = 80

  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443

  [[services.tcp_checks]]
    grace_period = "1s"
    interval = "15s"
    restart_limit = 6
    timeout = "2s"
```

**Command output:**

```text
Deploying icy-sky-8224
==> Validating app configuration
--> Validating app configuration done
Services
TCP 80/443 ⇢ 3000
==> Creating build context
--> Creating build context done
==> Building image with Docker
[+] Building 4.3s (2/2) FINISHED                                                                                                                                                                        
 => [internal] load remote build context                                                                                                                                                           0.0s
 => copy /context /                                                                                                                                                                                4.0s
Error error building: failed to solve with frontend dockerfile.v0: failed to read dockerfile: open /var/lib/docker/tmp/buildkit-mount710463774/Dockerfile: no such file or director
```

**Environment**

```
> flyctl version
flyctl v0.0.210 darwin/amd64 Commit: dfcc04f BuildDate: 2021-04-14T01:58:10Z

> docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.10
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:13:00 2021
 OS/Arch:           darwin/amd64 (rosetta)
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:16:48 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

> sw_vers
ProductName:    macOS
ProductVersion: 11.2.3
BuildVersion:   20D91
```

**Comment by elliotdickison** [comment]
If I move `./packages/client/Dockerfile` to `./Dockerfile` and re-run `flyctl deploy --config ./packages/client/fly.toml` without the `--dockerfile` flag then the app builds and deploys just fine, so I'm pretty sure this is an issue with the flag and not with my Dockerfile or toml config.

**Comment by nickysemenza** [comment]
I hit this on `v0.0.210` as well – I arbitrarily installed `v0.0.170` and that worked.

**Comment by elliotdickison** [comment]
Ah good catch! Will downgrade for now. Did some testing and the flag works for me through `v0.0.190` but breaks in `v0.0.192` (`v0.0.191` doesn't have artifacts for macOS). Tried looking through the changes between those two versions but didn't see anything obvious to my untrained eye: https://github.com/superfly/flyctl/compare/v0.0.190...v0.0.192. The "build and deploy refactor" change seems like a good place to start though...

**Comment by michaeldwan** [comment]
Sorry I missed this. The dockerfile issue was fixed in babb3333d0c724630d52c104e2c50fcab668660a.
